packages/map-template/src/components/Wayfinding/Wayfinding.jsx

```diff
@@ -404,16 +435,18 @@ function Wayfinding({ onStartDirections, onBack, directionsToLocation, direction
                             cleared={() => onSearchCleared(searchFieldIdentifiers.TO)}
                         />
                     </label>
-                    {userPosition && originLocation?.id !== 'USER_POSITION' && <p className="wayfinding__use-current-position">
-                        <button onClick={() => setMyPositionAsOrigin()}>Use My Position</button>
-                    </p>}
+
                 </div>
             </div>
             {!hasFoundRoute && <p className="wayfinding__error">No route found</p>}
             {!hasSearchResults && <p className="wayfinding__error">Nothing was found</p>}
             {searchResults.length > 0 &&
                 <div className="wayfinding__scrollable" {...scrollableContentSwipePrevent}>
                     <div className="wayfinding__results">
+                        {userPosition && !hasMyPostion && <div className="wayfinding__use-current-position" onClick={() => selectMyPosition()}>
+                            <CompassArrow />
+                            <span>My Position</span>
```

Comment:
Is the span necessary?

```suggestion
                            My Position
```